lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/grok/GrokDeserializer.java

```diff
@@ -0,0 +1,174 @@
+
+        // If line does not match grok pattern, return a row of nulls
+        if (match.getMatch() == null) {
+            for (int i = 0; i < columns.size(); i++) {
+                builder.getBlockBuilder(i).appendNull();
+            }
+            return;
+        }
+
+        try {
+            match.captures();
+        }
+        catch (GrokException e) {
+            throw new RuntimeException("Grok capture failure.", e);
+        }
+
+        Map<String, Object> map = match.toMap();
+        List<Object> row = new ArrayList<Object>(map.values());
+
+        for (int i = 0; i < columns.size(); i++) {
+            Column column = columns.get(i);
+            int ordinal = column.ordinal();
+            BlockBuilder blockBuilder = builder.getBlockBuilder(i);
+            String value = ordinal < row.size() ? String.valueOf(row.get(i)) : null;
+            if (value == null) {
+                blockBuilder.appendNull();
+                continue;
+            }
+            serializeValue(value, column, blockBuilder, this.grokNullOnParseError);
+        }
+    }
+
+    private static void serializeValue(String value, Column column, BlockBuilder builder, boolean grokNullOnParseError)
+    {
+        try {
+            Type type = column.type();
+            if (BOOLEAN.equals(type)) {
+                type.writeBoolean(builder, Boolean.parseBoolean(value));
+            }
+            else if (BIGINT.equals(type)) {
+                type.writeLong(builder, Long.parseLong(value));
+            }
+            else if (INTEGER.equals(type)) {
+                type.writeLong(builder, Integer.parseInt(value));
+            }
+            else if (SMALLINT.equals(type)) {
+                type.writeLong(builder, Short.parseShort(value));
+            }
+            else if (TINYINT.equals(type)) {
+                type.writeLong(builder, Byte.parseByte(value));
+            }
+            else if (REAL.equals(type)) {
+                type.writeLong(builder, floatToRawIntBits(Float.parseFloat(value)));
+            }
+            else if (DOUBLE.equals(type)) {
+                type.writeDouble(builder, Double.parseDouble(value));
+            }
+            else if (type instanceof VarcharType varcharType) {
+                type.writeSlice(builder, truncateToLength(Slices.utf8Slice(value), varcharType));
+            }
+            else if (type instanceof CharType charType) {
+                type.writeSlice(builder, truncateToLengthAndTrimSpaces(Slices.utf8Slice(value), charType));
+            }
+            else {
+                throw new TrinoException(NOT_SUPPORTED, "Unsupported column type: " + type);
+            }
+        }
+        catch (RuntimeException e) {
+            if (grokNullOnParseError) {
+                builder.appendNull();
+            }
+            else {
+                throw new RuntimeException(e);
```

Comment:
e is already an instance of `RuntimeException`, but there are problems:
- `TrinoException` is a subclass of `RuntimeException`, so the unsupported column type case is going to get caught here (which we don't want). You probably want to add a catch block explicitly for `TrinoException` where you just re-throw those.
- For other `RuntimeException` types, you probably want to re-throw those as a kind of `TrinoException` with an appropriate error code